hkmc2/shared/src/main/scala/hkmc2/semantics/Resolver.scala

```diff
@@ -396,12 +409,18 @@ class Resolver(tl: TraceLogger)
                 case N => ictx
                 case S(tpe) =>
                   // The symbol should be the duplicated one from
-                  // elaborator in order for the lowering stage to
-                  // correctly lower the program.
+                  // elaborator. This is because (the symbol of) the
+                  // class parameters are only accessible in the
+                  // constructor of the class-like definition in the
+                  // lowering stage. So, in the elaborator we duplicate
+                  // the class parametes into a private/public fields
+                  // (depending on if the parameter is modified by `val`
+                  // or not). We have to use the field symbols instead
+                  // of the parameter symbols here.
```

Comment:
Ok, I see what this is doing now. But this is a very ugly workaround; surely there's a better way of doing this... How about we just store the relevant information in the appropriate place, so we don't have to go fishing for it later?

Reply:
I.e., why doesn't the elaborator store the field symbol of the parameter to `Param` directly?